akvo/rsr/spa/app/modules/editor/section1/comp/project-picker.jsx

```diff
@@ -41,72 +60,66 @@ const ProjectPicker = ({ loading, projects, savedData, removeSetItem, hasImporte
+          <Field
+            name="externalParentIatiActivityId"
+            render={({ input }) => (
+              <FinalField
+                {...input}
+                placeholder={t('IATI Identifier')}
+                name="externalParentIatiActivityId"
+                control="input"
+              />
+            )}
           />
         )}
         {!isExternal && (
           <Field
-            name="relatedProjects[0].relatedProjectName"
-            render={(nameProps) => {
+            name="contributesToProject"
+            render={({ input }) => {
+              const $options =
+                projects && projects.length > 0
+                  ? ((input.value && state.searchStr.length === 0) ? [projects.find(it => it.id === input.value)].filter(it => it != null).map(({ id, title }) => ({ value: id, label: `[${id}] ${title ? title : t('Untitled project')}` })) : state.options)
+                  : [{ value: input.value, label: '' }]
               return (
-                <FinalField
-                  name="relatedProjects[0].relatedProject"
-                  render={({ input }) => {
-                    const $options =
-                      projects && projects.length > 0
-                        ? ((input.value && state.searchStr.length === 0) ? [projects.find(it => it.id === input.value)].filter(it => it != null).map(({ id, title }) => ({ value: id, label: `[${id}] ${title ? title : t('Untitled project')}` })) : state.options)
-                        : [{ value: input.value, label: input.value ? `[${input.value}] ${nameProps.input.value}` ? nameProps.input.value : `[${input.value}] ${t('Untitled project')}` : '' }]
-                    return (
-                      <Select
-                        {...input}
-                        value={input.value}
-                        onChange={(value) => { if(value != null) input.onChange(value); else removeItem(input, fields) }}
-                        loading={loading}
-                        showSearch
-                        onSearch={filterOptions}
-                        notFoundContent={state.loading ? <Spin size="small" /> : (state.searchStr.length === 0 ? <span>{t('Start typing...')}</span> : <span>{t('No results')}</span>)}
-                        filterOption={false}
-                        disabled={hasImportedResults}
-                      >
-                        {$options?.map(option => <Option value={option.value} key={option.value}>{option.label}</Option>)}
-                      </Select>
-                    )
-                  }}
-                />
+                <Select
+                  {...input}
+                  value={input.value}
+                  loading={loading}
+                  showSearch
+                  onSearch={filterOptions}
+                  notFoundContent={state.loading ? <Spin size="small" /> : (state.searchStr.length === 0 ? <span>{t('Start typing...')}</span> : <span>{t('No results')}</span>)}
```

Comment:
This is also too long and would benefit from linebreaks in order to be more readable.